client/src/components/FeedPost/index.tsx

```diff
@@ -3,64 +3,143 @@ import '../../styles/style.scss';
+        const {newsFeed:  {authorsOfLike: prevAuthorsOfLike}}: any = prevProps;
+        if (authorsOfLike !== prevAuthorsOfLike) {
+            this.props.setCountOfLikes(authorsOfLike.length);
+
+            const checkLoggedUserLikeExist = authorsOfLike.filter((userId: string) =>
+                this.props.user._id === userId,
+            );
+
+            this.props.checkUserLikeExist(!!checkLoggedUserLikeExist.length);
+        }
+    }
+
+    public onAddLike = (): void => {
+        const {
+            user: {_id: userId},
+            id: postId,
+        }: any = this.props;
+        const body = {userId, postId};
+        this.props.addLoggedUserLike(
+            this.props.loggedId,
+            postId,
+            this.props.authorsOfLike,
+        );
+        this.props.addLike(body);
+    }
+
+    public onDeleteLike = (): void => {
+        const {
+            user: {_id: userId},
+            id: postId,
+        }: any = this.props;
+        const body = {userId, postId};
+        this.props.removeLoggedUserLike(
+            this.props.loggedId,
+            postId,
+            this.props.authorsOfLike,
+        );
+        this.props.deleteLike(body);
+    }
 
     public render(): JSX.Element {
+        const {author, image, description, likeExist, authorsOfLike, id}: any = this.props;
+
+        const likeButton = likeExist ?
+            (<i className='fa fa-heart fa-lg pr-1 like' onClick={this.onDeleteLike}/>) :
+            (<i className='fa fa-heart-o fa-lg pr-1' onClick={this.onAddLike}/>);
+
         return (
             <div className='profile-post border mb-5'>
                 <div className='post-header p-2 border'>
                     <div className='d-flex flex-row'>
                         <img
-                            src={noAvatar}
+                            src={author.photoPath || noAvatar}
                             alt='avatar'
                             width={32}
                             height={32}
                             className='img-fluid rounded-circle'
                         />
                         <Link
-                            to={`/profile/${this.props.author}`}
+                            to={`/profile/${author.username}`}
                             className='text-dark mt-1 ml-3'>
-                            {this.props.author}
+                            {author.username}
                         </Link>
                     </div>
                 </div>
                 <img
-                    src={this.props.image}
+                    src={image}
                     className='w-100 img-fluid'
                     alt='post'
                 />
                 <div className='d-block mt-3 mb-2 pl-3'>
-                    <i className='fa fa-heart fa-2x fa-lg pr-1 like'/>
-                    <span className='pl-2'>5 likes</span>
+                    <span className='pl-2'>{likeButton} {authorsOfLike.length} likes</span>
```

Comment:
а чому authorsOfLike.length а не countOfLikes ?

Reply:
because countOfLikes doesnt work in this way, on the server we get data from profile post and authorofLikes it is an array of persons who added this like. And in this way we can get count of like